src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.netstandard.cs

```diff
@@ -36,6 +42,48 @@ internal AssemblyName GetAssemblyName(StringHandle nameHandle, Version version,
             return assemblyName;
         }
 
+        internal static unsafe AssemblyName GetAssemblyName(string assemblyFile)
+        {
+            FileStream? fileStream = null;
+            MemoryMappedFile? mappedFile = null;
+            MemoryMappedViewAccessor? accessor = null;
+            PEReader? peReader = null;
+            try
+            {
+                // Create stream because CreateFromFile(string, ...) uses FileShare.None which is too strict
+                fileStream = new FileStream(assemblyFile, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, false);
+                if (fileStream.Length == 0)
+                {
+                    throw new BadImageFormatException(SR.PEImageDoesNotHaveMetadata, assemblyFile);
+                }
+
+                mappedFile = MemoryMappedFile.CreateFromFile(
+                    fileStream, null, fileStream.Length, MemoryMappedFileAccess.Read, HandleInheritability.None, true);
+                accessor = mappedFile.CreateViewAccessor(0, 0, MemoryMappedFileAccess.Read);
+
+                SafeMemoryMappedViewHandle? safeBuffer = accessor.SafeMemoryMappedViewHandle;
+                peReader = new PEReader((byte*)safeBuffer.DangerousGetHandle(), (int)safeBuffer.ByteLength);
+                MetadataReader mdReader = peReader.GetMetadataReader(MetadataReaderOptions.None);
+                AssemblyName assemblyName = mdReader.GetAssemblyDefinition().GetAssemblyName();
+                return assemblyName;
+            }
+            catch (InvalidOperationException ex)
+            {
+                throw new BadImageFormatException(ex.Message);
+            }
+            finally
+            {
+                if (peReader != null)
```

Comment:
Nit: `peReader?.Dispose();`, etc.

Reply:
null-conditional operator was my first new language feature in C#6.0/Roslyn
I keep forgetting how handy it can be.